authors/apps/authentication/views.py

```diff
@@ -198,18 +198,20 @@ def post(self, request):
         # Query for email in database
         user = User.objects.filter(email=request.data['email']).first()
         if user is None:
-            return Response({"message": "The email you entered does not exist"})
+            return Response({"message": "The email you entered does not exist"}, status=status.HTTP_404_NOT_FOUND)
 
         # Get URL for client and include in the email for resetting password
-        current_site_domain = settings.CLIENT_URL
+        current_site_domain = request.META['HTTP_ORIGIN']
         # generate token
         token = default_token_generator.make_token(user)
 
         # Sends mail with url, path of reset password and token
-        mail_message = 'Dear ' + user.username + ',\n\n''We received a request to change your password on Authors Haven.\n\n' \
-                                                 'Click the link below to set a new password' \
-                                                 ' \n' + current_site_domain + '/reset-password/' + token + '/' \
-            '\n\nYours\n AuthorsHaven.'
+        mail_message = 'Dear {},\n\nWe received a request to change your \
+        password on Authors Haven.\n\nClick the link below to set a new \
+        password.\n{}/reset-password/{}\
+        \n\nYours\n AuthorsHaven.'.format(user.username,
+                                          current_site_domain, user.email)
```

Comment:
Where is the token? use query parameters @meshack-mbuvi

Reply:
Check the new changes